webapp/src/modules/order/sagas.ts

```diff
@@ -128,6 +132,11 @@ export function* orderSaga(tradeService: TradeService) {
       yield call(waitForFeatureFlagsToBeLoaded)
       const isOffchainPublicNFTOrdersEnabled: boolean = yield select(getIsOffchainPublicNFTOrdersEnabled)
       const wallet = (yield select(getWallet)) as ReturnType<typeof getWallet>
+
+      if (!wallet) {
+        throw new Error('A defined wallet is required to buy an item')
+      }
+
```

Comment:
Maybe if we move these checks over here, we could remove the duplicated code that checks for the credits to be available, what do you think?
```suggestion
if (useCredits) {
    const isCreditsEnabled: boolean = yield select(getIsCreditsEnabled)
    if (!isCreditsEnabled) {
      throw new Error('Credits are not enabled')
    }
    const credits: CreditsResponse = yield select(getCredits, wallet?.address || '')
    if (!credits || credits.totalCredits <= 0) {
      throw new Error('No credits available')
    }
}
```
As another option, we could create a function called `getCredits` which can do this and re-use it in the transak sagas file, which does the same checks.